### DISCUSSION TITLE
Image display possible

### DISCUSSION THREAD

**Discussion by Linwood-F** [Blackymas/NSPanel_HA_Blueprint#1270] [Q&A] [answered]
Is there any way to push an image, say when a doorbell rings, onto the device? 

Not video, just an image, e.g. have a page for it, push the image and invoke the page, maybe update it every 10s or some such, whatever it could handle.

Mine was going to go beside the front door for controlling landscape lights primarily, but it occurred to me it would be very nice to see the security camera (from which I can pull a jpg, e.g. on the picture card in HA at least I can) as I walk up to open it. 

Linwood

**Comment by edwardtfn** [comment]
Unfortunately the Nextion Display used by NSPanel don't have any component which allows image to be transferred programmatically. All the images must be added to the TFT file on the build.
It have, however, a way to plot pixel by pixel. Therefore this might be unusable for video, as we have limitations also on the serial interface, it should allow basic image plotting, specially with the 921600bps recently implemented with v4.1 of this project.
I was looking at the image library on ESPHome and it looks like it will provide some tools to help with this conversion, but I would have to investigate a bit to be able to say if this is viable or not.
I will try to find some time to go deep at this.

**Comment by Linwood-F** [comment]
Thank you. Video seems too much to hope for, but now that you opened that door a crack, that would be wonderful.

**Reply by edwardtfn** [reply, marked as answer]
In fact, ESPHome can't support not even dynamic images, so my hope is to try on implementation which works with images from a url: https://github.com/esphome/esphome/pull/4710
That would allow refreshing the image in an interval, but I doubt this would give something to be called video. It's a limitation from ESPHome in top of the limitations from Nextion.
We also have to deal with the limited memory available, which would propable force us to download and plot by chunks.
I'm not that optimistic about this request. 😩

**Reply by Linwood-F** [reply]
Thank you for the honest assessment; should I just mark that as an answer?

**Reply by Doughboy68** [reply]
A static image via URL would be neat...that opens up lots of possibilities. Eg. doorbell ring, snaps a picture and displays on panel... album art in media player, even those fancy AI generated images for weather in HA etc.

**Reply by edwardtfn** [reply]
Just an update on this...

I've been playing with a library to draw images on the Nextion device just like if it was a classic device for ESPHome.
I could kind of plot images, but as it have to be sent pixel by pixel and Nextion doesn't even have a command to plot a single pixel, it takes forever to draw (like more than a minute to plot a 480x320 image) and in 90% of my tries the system crashed before the image was complete.
For the crashes I'm pretty sure a solution is not that difficult, but for the performance I'm not very confident, unless we try to rebuild all ESPHome image libraries to be more efficient than a pixel by pixel algorytm, which I think is very unlikely to happen.

Sorry for the disappointing news.

😞

**Reply by Linwood-F** [reply]
Wow, thanks for the effort, that's disappointing but I certainly appreciate the work and the update.

**Comment by asantaga** [comment]
thanks for looking into this,
I think i understood that the nextation api only allows the showing of images which are already on the device? Is it possible with the esphome api to upload a file, and effectively replace a file already on the device? 
I need to look at the api and see how it can be used but I wonder if that trick was possible.. 

Tis a shame..  and I doubt the above would help but thought Id chime in
